src/service/auth.ts

```diff
@@ -63,17 +68,103 @@ export class AuthenticationService {
                 this.authToken = tokenFromFile;
             } else {
                 logger.info(`Access token not found in ${chalk.dim`ZEPLIN_ACCESS_TOKEN`} environment variable.`);
-                this.authToken = await this.promptForLogin();
+
+                const noBrowser = params?.noBrowser || false;
+
+                this.authToken = await this.promptForLogin({ ignoreSaveTokenErrors: true, noBrowser });
             }
         }
 
         return validate(this.authToken, params?.requiredScopes);
     }
 
-    async promptForLogin(
-        options: { ignoreSaveTokenErrors: boolean } = { ignoreSaveTokenErrors: true }
-    ): Promise<string> {
+    async promptForLogin(options: {
+        ignoreSaveTokenErrors: boolean; noBrowser: boolean;
+    } = {
+        ignoreSaveTokenErrors: true, noBrowser: false
+    }): Promise<string> {
         logger.info("\nLogin into Zeplin…");
+
+        let authToken: string | undefined;
+
+        if (!options.noBrowser) {
+            authToken = await this.promptForBrowserLogin();
+
+            if (!authToken) {
+                logger.info("Failed login using browser. Enter your login credentials below.");
+
+                authToken = await this.promptForCLILogin();
+            }
+        } else {
+            authToken = await this.promptForCLILogin();
+        }
+
+        authToken = validate(authToken);
+
+        try {
+            await authFileUtil.saveAuthToken(authToken);
+        } catch (err) {
+            logger.debug(`${err.stack}`);
+            if (!options.ignoreSaveTokenErrors) {
+                throw err;
+            }
+        }
+
+        return authToken;
+    }
+
+    async promptForBrowserLogin(): Promise<string | undefined> {
+        const authUrl = new URL("/oauth/authorize", defaults.app.webURL);
+
+        authUrl.searchParams.append("client_id", defaults.api.clientId);
+        authUrl.searchParams.append("scope", "read+write+delete");
```

Comment:
```suggestion
        authUrl.searchParams.append("client_id", defaults.api.clientId);
        authUrl.searchParams.append("scope", "read+write+delete");
        authUrl.searchParams.append("redirect_uri", defaults.api.webAuthRedirectURL);
```

`defaults.api.webAuthRedirectURL` should be `process.env.WEB_AUTH_REDIRECT_URI || "http://app.zeplin.io/cliAuthorized"`